cli/util.go

```diff
@@ -307,6 +308,10 @@ func natsOpts() []nats.Option {
 		connectionName = "NATS CLI Version " + Version
 	}
 
+	if opts().TlsInsecure {
+		copts = append(copts, nats.Secure(&tls.Config{InsecureSkipVerify: true}))
```

Comment:
the problem with this approach is that it invalidates all other tls setup.  You may very well still want to send certificate and use a key but you just want to skip the CA setup.

This approach will only work with servers that dont validate your client cert.  I've previously rejected essentially this same PR because I dont think we could come up with a plan that worked better.


We would essentially have to create an entire valid tlsc with all things loaded just like the client would with just this one setting set in addition.

I think the go client should get a feature flag to enable this instead

Reply:
I think TLS Insecure is just for usecases where the server also has insecure flag set and doesnt validate tls ceritificates. 
Instead of creating a new tls.Config{} we could use the existing tls.Config if it would be exported somewhere.

Reply:
Yeah, I think better if we get an option into nats.go to set this flag

Reply:
Its better to have this option with some exeptions like it doesnt support sending certificates instead of not able to connect any insecure tls server. The fix for that issue could be added later on when the go client supports it.

Reply:
I disagree a bit I am not conformable doing essentially just YOLO TLS as the only option.

Reply:
Ok i found another solution to this, please check my new commit